pkg/controller/plan/adapter/vsphere/validator.go

```diff
@@ -127,3 +127,29 @@ func (r *Validator) MaintenanceMode(vmRef ref.Ref) (ok bool, err error) {
 func (r *Validator) DirectStorage(vmRef ref.Ref) (bool, error) {
 	return true, nil
 }
+
+// Validate that every VM NIC has it's data from the guest to get static IPs
+func (r *Validator) StaticIPs(vmRef ref.Ref) (ok bool, err error) {
+	if !r.plan.Spec.PreserveStaticIPs {
+		return true, nil
+	}
+	vm := &model.Workload{}
+	err = r.inventory.Find(vm, vmRef)
+	if err != nil {
+		err = liberr.Wrap(err, "vm", vmRef.String())
+		return
+	}
+	for _, nic := range vm.NICs {
+		found := false
```

Comment:
I think the code within this for-loop would be [better with a label](https://relistan.com/continue-statement-with-labels-in-go), but if you have something against labels then at least we can break when we find a corresponding guestNetwork for the NIC..

Reply:
do we ever used the labels in the project? i'm unsure about my opinion to add `goto` in the code

Reply:
apparently, [we do](https://github.com/kubev2v/forklift/blob/main/pkg/controller/provider/container/vsphere/collector.go#L300)

Reply:
i prefer it as is, but i might need to verify again the PR after the last change